### PR TITLE
feat(backend): improve log for http error response

### LINKED ISSUE #442: When a timeout error occurs, the error message is not clear enough.

### PR DESCRIPTION
**Pull request by LiteSun**
## Summary

Fixes #442

When the API7 Dashboard returns non-2xx HTTP responses with empty body (e.g., HTTP 500 under high concurrency), ADC displayed `Error: ""` which provides no useful diagnostic information.

## Root Cause Analysis

The bug is in the `catchError` handler in `operator.ts` across all three backends:

```typescript
new Error(
  error.response?.data?.error_msg ??
    JSON.stringify(error.response?.data),
)
```

When the Dashboard returns HTTP 500 with empty body:
- `error.response.data` = `""` (empty string)
- `"".error_msg` = `undefined`
- `JSON.stringify("")` = `'""'` (the string `""`)
- Result: `Error: ""`

This happens because:
1. The Dashboard can return 500 + empty body under high concurrency (verified via real API calls)
2. The `??` (nullish coalescing) only checks `null`/`undefined`, not empty string
3. `JSON.stringify("")` produces the misleading `""` output

### Why 499 appears in Dashboard logs

When ADC sync runs with high concurrency (e.g., `--request-concurrent 1000`) and one request fails:
1. `exitOnFailure: true` causes `throwError()` in the rxjs chain
2. `mergeMap` tears down all inner observables
3. Remaining ~999 in-flight HTTP connections are closed
4. Dashboard Nginx logs each client disconnect as **499**

The 499s are a **side effect** of the first error, not the cause.

### 1. `formatAxiosErrorMessage()` in SDK utils (`libs/sdk/src/utils.ts`)

New shared function that formats axios error responses with:
- HTTP method and URL
- Response status code and status text
- `error_msg` from response body (if available)
- Raw response body (when no `error_msg` field exists)

**Before:** `Error: ""`
**After:** `Error: PUT https://dashboard.example.com/apisix/admin/routes/123, responded with status 500 Internal Server Error`

### 2. Applied to all three backends

Updated `catchError` in `operator.ts` for:
- `libs/backend-api7/src/operator.ts`
- `libs/backend-apisix/src/operator.ts`
- `libs/backend-apisix-standalone/src/operator.ts`

Both `exitOnFailure` and non-`exitOnFailure` paths are fixed.

### 3. Timeout error improvement (previous commits)

Added `registerTimeoutInterceptor` that enhances timeout errors:

**Before:** `AxiosError: timeout of 5000ms exceeded` (no URL/method info)
**After:** `Request "GET https://example.com/api/version" timed out after 5000ms. Consider increasing the timeout with the --timeout flag.`

## Tests

- 6 new unit tests for `formatAxiosErrorMessage` covering:
  - Normal error with `error_msg`
  - Empty body response (the `Error: ""` scenario)
  - HTML response body (e.g., Nginx 502)
  - Absolute URL handling
  - Missing config graceful handling
  - JSON response without `error_msg`
- 5 existing unit tests for `registerTimeoutInterceptor`
- 4 existing integration tests for timeout scenarios

**Test coverage note:** The `Error: ""` scenario requires the Dashboard to return HTTP 500 with empty body, which occurs under specific high-concurrency conditions on the same resource. This was verified through real API calls (300 concurrent PUTs to the same route ID produced 63% HTTP 500 + empty body responses). The fix is defensive and handles all response body formats correctly.

## Release Notes

* **Improvements**
  * Timeout errors now include request method, URL, and timeout duration to aid debugging and troubleshooting
  * Enhanced error message formatting and consistency across all API operations

* **Tests**
  * Added timeout error handling tests to validate error message accuracy and content

[Review Change Stack](https://app.coderabbit.ai/change-stack/api7/adc/pull/443)

### CODE REVIEW

File: libs/sdk/src/utils.ts [hunk 1]
```diff
@@ -28,9 +29,29 @@ const featureGateEnabled = (key: string) => {
   );
 };
 
+const registerTimeoutInterceptor = (client: AxiosInstance) => {
+  client.interceptors.response.use(undefined, (error) => {
+    if (axios.isAxiosError(error) && error.code === 'ECONNABORTED') {
+      const method = error.config?.method?.toUpperCase() ?? 'UNKNOWN';
+      const url = `${error.config?.baseURL ?? ''}${error.config?.url ?? ''}`;
+      const timeout = error.config?.timeout;
+      const newMessage = `Request "${method} ${url}" timed out after ${timeout}ms. Consider increasing the timeout with the --timeout flag.`;
```
[the remark this answers is not in the document]
Comment (LiteSun) [reply libs/sdk/src/utils.ts]: Good catch! Fixed in e472ecf:
- Absolute URLs (starting with `http://` or `https://`) are now used directly without prepending `baseURL`.
- Missing timeout values now display `an unknown duration` instead of `undefinedms`.
- Added test cases for both edge cases.

**Review by bzp2010 — APPROVED** [review]
